### PR TITLE
LPS-181346 | Automation FDSViews#CanAssertPaginationOnFDSViewsPage

### PR DESCRIPTION
**Pull request by abhnerramos**
**Description**
Automation for FDSViews#CanAssertPaginationOnFDSViewsPage - LPS-181346
Automation for FDSViews#AssertNumberOfCharactersExceededInNameField - LPS-181348 (already checked)
Automation for FDSViews#CanSearchViewByName - LPS-181351
Automation FDSViews#CanCancelViewCreationOnCancelButton - LPS-181342
Automation for FDSViews#CanCreateViewWithNoDescription - LPS-181341

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-181346
https://issues.liferay.com/browse/LPS-181348
https://issues.liferay.com/browse/LPS-181351
https://issues.liferay.com/browse/LPS-181342
https://issues.liferay.com/browse/LPS-181341

### CODE REVIEW

**Comment by abhnerramos** [comment]
ci:stop

**Comment by abhnerramos** [comment]
ci:test:frontend

**Comment by abhnerramos** [comment]
hey @mateusralv , LGTM! Could you take a look at it, please?

**Comment by abhnerramos** [comment]
ci:test:sf

**Comment by abhnerramos** [comment]
ci:test:frontend

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro [hunk 1]
```diff
@@ -19,17 +19,36 @@ definition {
 	}
 
 	macro createDataSetView {
-		LexiconEntry.gotoAdd();
+		var reps = ${reps};
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro:22]:
```suggestion

```

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro [hunk 2]
```diff
@@ -19,17 +19,36 @@ definition {
...
 
-		Type(
-			locator1 = "TextInput#NAME",
-			value1 = ${key_name});
+		if (isSet(reps)) {
+			for (var num : list ${reps}) {
+				LexiconEntry.gotoAdd();
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro:26]:
```suggestion
		if (isSet(key_dataSetViewNameList)) {
			for (var key_dataSetViewName : list ${key_dataSetViewNameList}) {
				LexiconEntry.gotoAdd();
```

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro [hunk 3]
```diff
@@ -19,17 +19,36 @@ definition {
...
 
-		Type(
-			locator1 = "FrontendDataSetAdmin#NEW_DATA_SET_VIEW_DESCRIPTION",
-			value1 = ${description});
+				Type(
+					locator1 = "TextInput#NAME",
+					value1 = "${key_name} ${num}");
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro:30]:
```suggestion
				Type(
					locator1 = "TextInput#NAME",
					value1 = "${key_dataSetViewName}");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase [hunk 4]
```diff
@@ -34,6 +34,55 @@ definition {
...
+		task ("When go to the View page of the dataset created.") {
+			FrontendDataSetAdmin.goToViews(dataSetName = "Dataset Test");
+		}
+
+		task ("And Adding a 6 view called 'View Test 1', 'View Test 2'... and Filling the description field with 'Test Description'") {
+			FrontendDataSetAdmin.createDataSetView(
+				description = "Test Description",
+				key_name = "View Test",
+				reps = "1,2,3,4,5,6");
+		}
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase:55]:
```suggestion
			FrontendDataSetAdmin.createDataSetView(
				description = "Test Description",
				key_dataSetViewNameList = "View Test 1,View Test 2,View Test 3,View Test 4,View Test 5,View Test 6 ");
		}
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase [hunk 5]
```diff
@@ -34,6 +34,55 @@ definition {
...
+
+		task ("And changing the page to 4 entries.") {
+			FrontendDataSetAdmin.changePagination(itemsPerPage = "4 Items");
+		}
+
+		task ("Then confirm that only 4 entries are displayed") {
+			for (var num : list "1,2,3,4") {
+				AssertElementPresent(
+					entryName = "View Test ${num}",
+					locator1 = "FrontendDataSetAdmin#DATA_SET_VIEW_ENTRY_NAME");
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase:65]: Could you create a macro similar to the one you did above to put this for inside?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase [hunk 6]
```diff
@@ -34,6 +34,55 @@ definition {
...
+
+		task ("When to switch to page 2") {
+			Click(locator1 = "Pagination#NEXT_LINK");
+		}
+
+		task ("Then 2 entries are displayed") {
+			for (var num : list "5,6") {
+				AssertElementPresent(
+					entryName = "View Test ${num}",
+					locator1 = "FrontendDataSetAdmin#DATA_SET_VIEW_ENTRY_NAME");
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/datamanagment/FDSViews.testcase:79]: Use the macro created here as well.

**Review by mateusralv — APPROVED** [review]
Hi @abhnerramos You did a great job with the macro to create views, I only requested minor changes related to variable naming, other than that the macro is working great. Congratulations 🎉

**Comment by abhnerramos** [comment]
ci:test:sf

**Comment by abhnerramos** [comment]
ci:test:frontend-data-set

**Comment by abhnerramos** [comment]
To avoid conflicts in this branch, I'll be merging this test into #3250

**Comment by abhnerramos** [comment]
ci:open

**Comment by abhnerramos** [comment]
ci:test:sf

**Comment by abhnerramos** [comment]
ci:test:frontend-data-set

**Comment by mateusralv** [comment]
ci:test:frontend

**Review by mateusralv — APPROVED** [review]
The tests are passing on CI. Good job! see: [ci:test:frontend](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2273279688)

**Comment by mateusralv** [comment]
ci:forward

**Pull request closed without merging**